src/Messages/TSMessagesManager.m

```diff
@@ -155,6 +155,15 @@ - (void)handleReceivedEnvelope:(OWSSignalServiceProtosEnvelope *)envelope
     OWSAssert([NSThread isMainThread]);
 
     DDLogInfo(@"%@ received envelope: %@", self.tag, [self descriptionForEnvelope:envelope]);
+
+    // TODO: Can we trust envelope.source to be properly formatted?
```

Comment:
re: TODO 

Per previous discussion, this is a server provided value. As long as we continue to  use the same `source` to fetch the session identifier, any spoofed value here would not be decryptable.

We'd have to revisit this upon allowing non-phone identifiers.